### PR TITLE
[Feature][Connector-V2]Support tidb cdc connector source

### LINKED ISSUE #7199: [Feature][CDC] Support tidb cdc connector source
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Currently, we don't support reading tidb change data capture (CDC), does anyone want to add support for tidb?

docs
https://github.com/tikv/client-java

reference
https://github.com/apache/seatunnel/tree/dev/seatunnel-connectors-v2/connector-cdc/connector-cdc-mysql
https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e
https://github.com/apache/seatunnel/tree/dev/docs

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by sunxiaojian**
### Purpose of this pull request

closed https://github.com/apache/seatunnel/issues/7199

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: .idea/vcs.xml [hunk 1]
```diff
@@ -1,24 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<!--
```
Comment (hailin0) [inline .idea/vcs.xml]: why update this file

File: docs/en/connector-v2/source/TiDB-CDC.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
+# PostgreSQL CDC
```
Comment (hailin0) [inline docs/en/connector-v2/source/TiDB-CDC.md]: Please add zh docs

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/mysql/MySqlTypeConverter.java [hunk 3]
```diff
@@ -140,7 +142,9 @@ public Column convert(BasicTypeDefine typeDefine) {
             case MYSQL_NULL:
                 builder.dataType(BasicType.VOID_TYPE);
                 break;
+
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/mysql/MySqlTypeConverter.java]: revert

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-tidb-e2e/src/test/java/org/apache/seatunnel/e2e/connector/tidb/TiDBCDCIT.java [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+import java.util.concurrent.TimeUnit;
+
+import static org.awaitility.Awaitility.await;
+
+@Slf4j
+@DisabledOnContainer(
+        value = {},
+        type = {EngineType.SPARK},
+        disabledReason = "Currently SPARK do not support cdc")
+public class TiDBCDCIT extends TiDBTestBase implements TestResource {
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-tidb-e2e/src/test/java/org/apache/seatunnel/e2e/connector/tidb/TiDBCDCIT.java:54]: reference mysql e2e

https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/resources

File: docs/zh/connector-v2/source/TiDB-CDC.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+> TiDB CDC模式的连接器
+
+## 支持的引擎
+
+> SeaTunnel Zeta<br/>
+> Flink <br/>
+
+## 主要功能
+
+- [x] [批处理](../../concept/connector-v2-features.md)
```
Comment (hailin0) [inline docs/zh/connector-v2/source/TiDB-CDC.md]: support batch mode？

**Comment by hailin0** [comment]
cc @dailai

**Comment by dailai** [comment]
Please resolve the confilict and retrigger the ci.

**Comment by dailai** [comment]
Please fix the ci.

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-tidb-e2e/src/test/java/org/apache/seatunnel/e2e/connector/tidb/TiDBCDCIT.java [hunk 6]
```diff
@@ -0,0 +1,372 @@
...
```
Comment (dailai) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-tidb-e2e/src/test/java/org/apache/seatunnel/e2e/connector/tidb/TiDBCDCIT.java:54]: Thanks @sunxiaojian. Please add the e2e case that restore from savepoint.

**Review by dailai — APPROVED** [review]
LGTM

File: docs/en/connector-v2/source/TiDB-CDC.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+| database-name                | List    | Yes      | -       | Database name of the database to monitor.                                                                                                                                                                                                                                                                                                                                                    |
+| table-name                   | List    | Yes      | -       | Table name of the database to monitor. The table name needs to include the database name.                                                                                                                                                                                                                                                                                                    |
```
Comment (hailin0) [inline docs/en/connector-v2/source/TiDB-CDC.md]:
```suggestion
| table-names                   | List    | Yes      | -       | Table name of the database to monitor. The table name needs to include the database name.                                                                                                                                                                                                                                                                                                    |
```

Comment (hailin0) [reply docs/en/connector-v2/source/TiDB-CDC.md]: Please use array configuration on example or e2e

Comment (sunxiaojian) [reply docs/en/connector-v2/source/TiDB-CDC.md]: > Please use array configuration on example or e2e

Multiple tables are not supported now,  Upgrade in the next version

**Review by hailin0 — APPROVED** [review]
LGTM

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**